diff --git a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
index e2500e31c..4a1dbbe05 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
@@ -8,6 +8,7 @@
 #include <dt-bindings/interrupt-controller/irq.h>
 #include <dt-bindings/phy/phy.h>
 #include <dt-bindings/power/rk3588-power.h>
+#include <dt-bindings/soc/rockchip,boot-mode.h>
 #include <dt-bindings/reset/rockchip,rk3588-cru.h>
 #include <dt-bindings/phy/phy.h>
 #include <dt-bindings/ata/ahci.h>
@@ -767,6 +768,28 @@
 		#reset-cells = <1>;
 	};
 
+	pmu0_grf: syscon@fd588000 {
+    	compatible = "rockchip,rk3588-pmu0-grf", "syscon", "simple-mfd";
+    	reg = <0x0 0xfd588000 0x0 0x2000>;
+    
+    	reboot_mode: reboot-mode {
+    		compatible = "syscon-reboot-mode";
+    		offset = <0x80>;
+    		mode-bootloader = <BOOT_BL_DOWNLOAD>;
+    		mode-charge = <BOOT_CHARGING>;
+    		mode-fastboot = <BOOT_FASTBOOT>;
+    		mode-loader = <BOOT_BL_DOWNLOAD>;
+    		mode-normal = <BOOT_NORMAL>;
+    		mode-recovery = <BOOT_RECOVERY>;
+    		mode-ums = <BOOT_UMS>;
+    		mode-panic = <BOOT_PANIC>;
+    		mode-watchdog = <BOOT_WATCHDOG>;
+    		mode-quiescent = <BOOT_QUIESCENT>;
+    		/* add a mode to capture the ramdump through usb */
+    		mode-winusb = <BOOT_WINUSB>;
+    	};
+    };
+
 	i2c0: i2c@fd880000 {
 		compatible = "rockchip,rk3588-i2c", "rockchip,rk3399-i2c";
 		reg = <0x0 0xfd880000 0x0 0x1000>;
@@ -1366,6 +1389,18 @@
 		resets = <&cru SRST_A_AV1>, <&cru SRST_P_AV1>, <&cru SRST_A_AV1_BIU>, <&cru SRST_P_AV1_BIU>;
 	};
 
+	av1d_mmu: iommu@fdca0000 {
+		compatible = "rockchip,iommu-av1d";
+		reg = <0x0 0xfdca0000 0x0 0x600>;
+		interrupts = <GIC_SPI 109 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "irq_av1d_mmu";
+		clocks = <&cru ACLK_AV1>, <&cru PCLK_AV1>;
+		clock-names = "aclk", "iface";
+		#iommu-cells = <0>;
+		power-domains = <&power RK3588_PD_AV1>;
+		status = "disabled";
+	};
+
 	vop: vop@fdd90000 {
 		compatible = "rockchip,rk3588-vop";
 		reg = <0x0 0xfdd90000 0x0 0x4200>, <0x0 0xfdd95000 0x0 0x1000>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-compact.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-compact.dtsi
index 9343dfc86..09f0e85a2 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-compact.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-compact.dtsi
@@ -19,6 +19,38 @@
 		regulator-max-microvolt = <5000000>;
 		vin-supply = <&vcc5v0_sys>;
 	};
+
+	sdio_pwrseq: sdio-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		clocks = <&hym8563>;
+		clock-names = "ext_clock";
+	    pinctrl-names = "default";
+	    pinctrl-0 = <&wifi_enable_h>;
+	    /*
+	    * On the module itself this is one of these (depending
+	    * on the actual card populated):
+	    * - SDIO_RESET_L_WL_REG_ON
+	    * - PDN (power down when low)
+	    */
+	    post-power-on-delay-ms = <200>;
+		reset-gpios = <&gpio2 RK_PC5 GPIO_ACTIVE_LOW>;
+		status = "okay";
+	};
+
+	wireless_wlan: wireless-wlan {
+        compatible = "android,bcmdhd_wlan";
+		reg = <1>;
+        wifi_chip_type = "ap6611";
+        pinctrl-names = "default";
+        pinctrl-0 = <&wifi_host_wake_irq>;
+		wl_reg_on-gpios = <&gpio2 RK_PC5 GPIO_ACTIVE_HIGH>;
+        wl_host_wake-gpios = <&gpio0 RK_PB0 GPIO_ACTIVE_HIGH>;
+        interrupt-parent = <&gpio0>;
+        interrupts = <RK_PB0 IRQ_TYPE_LEVEL_HIGH>;
+        interrupt-names = "host-wake";
+        status = "okay";
+    };
+
 };
 
 &headphone_amp {
@@ -101,6 +133,16 @@
 			rockchip,pins = <4 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	wireless-wlan {
+		wifi_enable_h: wifi-enable-h {
+			rockchip,pins = <2 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		wifi_host_wake_irq: wifi-host-wake-irq {
+			rockchip,pins = <0 RK_PB0 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
+	};
 };
 
 &pwm4 {
@@ -176,3 +218,39 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&usb_host_pwren>;
 };
+
+&sdio {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	bus-width = <4>;
+	cap-sdio-irq;                   
+    cap-sd-highspeed;
+	keep-power-in-suspend;    
+    disable-wp; 
+	sd-uhs-sdr104;
+    pwrseq-1.8v;
+	max-frequency = <150000000>;
+	mmc-pwrseq = <&sdio_pwrseq>;
+	non-removable;
+	no-mmc;
+	no-sd;
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdiom0_pins>;
+	status = "okay";
+};
+
+&sdiom0_pins {
+	rockchip,pins =
+		/* sdio_clk_m0 */
+		<2 RK_PB3 2 &pcfg_pull_none>,
+		/* sdio_cmd_m0 */
+		<2 RK_PB2 2 &pcfg_pull_up>,
+		/* sdio_d0_m0 */
+		<2 RK_PA6 2 &pcfg_pull_up>,
+		/* sdio_d1_m0 */
+		<2 RK_PA7 2 &pcfg_pull_up>,
+		/* sdio_d2_m0 */
+		<2 RK_PB0 2 &pcfg_pull_up>,
+		/* sdio_d3_m0 */
+		<2 RK_PB1 2 &pcfg_pull_up>;
+};
\ No newline at end of file
diff --git a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-max.dts b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-max.dts
index 8b1d35760..78f1048f0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-max.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-max.dts
@@ -32,6 +32,133 @@
 			};
 		};
 	};
+
+	rkvenc_ccu: rkvenc-ccu {
+		compatible = "rockchip,rkv-encoder-v2-ccu";
+		status = "okay";
+	};
+
+	mpp_srv: mpp-srv {
+		compatible = "rockchip,mpp-service";
+		rockchip,taskqueue-count = <12>;
+		rockchip,resetgroup-count = <1>;
+		status = "okay";
+	};
+
+	rkvenc0: rkvenc-core@fdbd0000 {
+		compatible = "rockchip,rkv-encoder-v2-core";
+		reg = <0x0 0xfdbd0000 0x0 0x6000>;
+		interrupt-parent = <&gic>;
+		interrupts = <&gic 101 IRQ_TYPE_LEVEL_HIGH 0>;
+		interrupt-names = "irq_rkvenc0";
+		clocks = <&cru ACLK_RKVENC0>, <&cru HCLK_RKVENC0>, <&cru CLK_RKVENC0_CORE>;
+		clock-names = "aclk_vcodec", "hclk_vcodec", "clk_core";
+		rockchip,normal-rates = <500000000>, <0>, <800000000>;
+		assigned-clocks = <&cru ACLK_RKVENC0>, <&cru CLK_RKVENC0_CORE>;
+		assigned-clock-rates = <500000000>, <800000000>;
+		resets = <&cru SRST_A_RKVENC0>, <&cru SRST_H_RKVENC0>, <&cru SRST_RKVENC0_CORE>;
+		reset-names = "video_a", "video_h", "video_core";
+		rockchip,skip-pmu-idle-request;
+		iommus = <&rkvenc0_mmu>;
+		rockchip,srv = <&mpp_srv>;
+		rockchip,ccu = <&rkvenc_ccu>;
+		rockchip,taskqueue-node = <7>;
+		rockchip,task-capacity = <8>;
+		power-domains = <&power RK3588_PD_VENC0>;
+		operating-points-v2 = <&venc_opp_table>;
+		status = "okay";
+	};
+
+	rkvenc0_mmu: iommu@fdbdf000 {
+		compatible = "rockchip,rk3568-iommu";
+		reg = <0x0 0xfdbdf000 0x0 0x40>, <0x0 0xfdbdf040 0x0 0x40>;
+		interrupt-parent = <&gic>;
+	interrupts = <GIC_SPI 99 IRQ_TYPE_LEVEL_HIGH 0>,
+			 <GIC_SPI 100 IRQ_TYPE_LEVEL_HIGH 0>;
+		interrupt-names = "irq_rkvenc0_mmu0", "irq_rkvenc0_mmu1";
+		clocks = <&cru ACLK_RKVENC0>, <&cru HCLK_RKVENC0>;
+		clock-names = "aclk", "iface";
+		rockchip,disable-mmu-reset;
+		rockchip,enable-cmd-retry;
+		rockchip,shootdown-entire;
+		#iommu-cells = <0>;
+		power-domains = <&power RK3588_PD_VENC0>;
+		status = "okay";
+	};
+
+	rkvenc1: rkvenc-core@fdbe0000 {
+		compatible = "rockchip,rkv-encoder-v2-core";
+		reg = <0x0 0xfdbe0000 0x0 0x6000>;
+		interrupt-parent = <&gic>;
+		interrupts = <GIC_SPI 104 IRQ_TYPE_LEVEL_HIGH 0>;
+		interrupt-names = "irq_rkvenc1";
+		clocks = <&cru ACLK_RKVENC1>, <&cru HCLK_RKVENC1>, <&cru CLK_RKVENC1_CORE>;
+		clock-names = "aclk_vcodec", "hclk_vcodec", "clk_core";
+		rockchip,normal-rates = <500000000>, <0>, <800000000>;
+		assigned-clocks = <&cru ACLK_RKVENC1>, <&cru CLK_RKVENC1_CORE>;
+		assigned-clock-rates = <500000000>, <800000000>;
+		resets = <&cru SRST_A_RKVENC1>, <&cru SRST_H_RKVENC1>, <&cru SRST_RKVENC1_CORE>;
+		reset-names = "video_a", "video_h", "video_core";
+		rockchip,skip-pmu-idle-request;
+		iommus = <&rkvenc1_mmu>;
+		rockchip,srv = <&mpp_srv>;
+		rockchip,ccu = <&rkvenc_ccu>;
+		rockchip,taskqueue-node = <7>;
+		rockchip,task-capacity = <8>;
+		power-domains = <&power RK3588_PD_VENC1>;
+		operating-points-v2 = <&venc_opp_table>;
+		status = "okay";
+	};
+
+	rkvenc1_mmu: iommu@fdbef000 {
+		compatible = "rockchip,rk3568-iommu";
+		reg = <0x0 0xfdbef000 0x0 0x40>, <0x0 0xfdbef040 0x0 0x40>;
+		interrupt-parent = <&gic>;
+	interrupts = <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH 0>,
+			 <GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH 0>;
+		interrupt-names = "irq_rkvenc1_mmu0", "irq_rkvenc1_mmu1";
+		clocks = <&cru ACLK_RKVENC1>, <&cru HCLK_RKVENC1>;
+		clock-names = "aclk", "iface";
+		rockchip,disable-mmu-reset;
+		rockchip,enable-cmd-retry;
+		rockchip,shootdown-entire;
+		#iommu-cells = <0>;
+		power-domains = <&power RK3588_PD_VENC1>;
+		status = "okay";
+	};
+
+	venc_opp_table: venc-opp-table {
+		compatible = "operating-points-v2";
+
+		nvmem-cells = <&codec_leakage>, <&venc_opp_info>;
+		nvmem-cell-names = "leakage", "opp-info";
+		rockchip,leakage-voltage-sel = <
+			1	15	0
+			16	25	1
+			26	254	2
+		>;
+
+		rockchip,grf = <&sys_grf>;
+		volt-mem-read-margin = <
+			855000	1
+			765000	2
+			675000	3
+			495000	4
+		>;
+
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <750000 750000 850000>,
+					<750000 750000 850000>;
+			opp-microvolt-L0 = <800000 800000 850000>,
+					   <800000 800000 850000>;
+			opp-microvolt-L1 = <775000 775000 850000>,
+					   <775000 775000 850000>;
+			opp-microvolt-L2 = <750000 750000 850000>,
+					   <750000 750000 850000>;
+		};
+	};
+
 };
 
 &hdmi0 {
@@ -123,3 +250,119 @@
 		remote-endpoint = <&hdmi1_in_vp1>;
 	};
 };
+
+&vpu121 {
+	vpu-supply = <&vdd_vdenc_s0>;
+	status = "okay";
+};
+
+&vpu121_mmu {
+	status = "okay";
+};
+
+&av1d {
+	vpu-supply = <&vdd_vdenc_s0>;
+	status = "okay";
+};
+
+&av1d_mmu {
+	status = "okay";
+};
+
+&vepu121_0 {
+	vpu-supply = <&vdd_vdenc_s0>;
+	status = "okay";
+};
+
+&vepu121_0_mmu {
+	status = "okay";
+};
+
+&vepu121_1 {
+	vpu-supply = <&vdd_vdenc_s0>;
+	status = "okay";
+};
+
+&vepu121_1_mmu {
+	status = "okay";
+};
+
+&vepu121_2 {
+	vpu-supply = <&vdd_vdenc_s0>;
+	status = "okay";
+};
+
+&vepu121_2_mmu {
+	status = "okay";
+};
+
+&vepu121_3 {
+	vpu-supply = <&vdd_vdenc_s0>;
+	status = "okay";
+};
+
+&vepu121_3_mmu {
+	status = "okay";
+};
+
+&pmu {
+	pmu_io_domains: pmu-io-domains {
+		compatible = "rockchip,rk3588-pmu-io-domains";
+		pmuio1-supply = <&vcc_1v8_s0>;
+		pmuio2-supply = <&vcc_1v8_s0>;
+		vccio1-supply = <&avcc_1v8_s0>;
+		vccio2-supply = <&vcc_1v8_s0>;
+		vccio3-supply = <&vcc_3v3_s0>;
+		vccio4-supply = <&vcc_1v8_s0>;
+		vccio5-supply = <&vcc_3v3_s0>;
+		vccio6-supply = <&vcc_1v8_s0>;
+		vccio7-supply = <&vcc_3v3_s0>;
+		status = "okay";
+	};
+};
+
+&i2c1 {
+	pinctrl-0 = <&i2c1m4_xfer>;
+	status = "disabled"; 
+};
+
+&spi2 {
+    pmic@0 {
+        pinctrl-0 = <&pmic_pins>;
+    };
+};
+
+&sfc {
+	spi_flash: flash@0 {
+		vcc-supply = <&vcc_3v3_s3>;
+	};
+};
+
+ &pd_gpu {
+	/delete-property/ domain-supply;
+ };
+ 
+ &pd_npu {
+	/delete-property/ domain-supply;
+ };
+ 
+ &rknn_core_0 {
+	/delete-property/ npu-supply;
+	/delete-property/ sram-supply;
+ };
+ 
+ &rknn_core_1 {
+	/delete-property/ npu-supply;
+	/delete-property/ sram-supply;
+ };
+ 
+ &rknn_core_2 {
+	/delete-property/ npu-supply;
+	/delete-property/ sram-supply;
+ };
+
+ &otp{
+	venc_opp_info: venc-opp-info@67 {
+		reg = <0x67 0x6>;
+	};
+ };
diff --git a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi
index 3bceee948..2fde9ba8a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5.dtsi
@@ -16,6 +16,7 @@
 	aliases {
 		mmc0 = &sdhci;
 		mmc1 = &sdmmc;
+		mmc2 = &sdio;
 	};
 
 	chosen {
@@ -99,13 +100,6 @@
 		};
 	};
 
-	rfkill {
-		compatible = "rfkill-gpio";
-		label = "rfkill-pcie-wlan";
-		radio-type = "wlan";
-		shutdown-gpios = <&gpio0 RK_PC4 GPIO_ACTIVE_HIGH>;
-	};
-
 	analog_sound: sound {
 		compatible = "simple-audio-card";
 		simple-audio-card,name = "Analog";
@@ -481,10 +475,10 @@
 		vcc8-supply = <&vcc5v0_sys>;
 		vcc9-supply = <&vcc5v0_sys>;
 		vcc10-supply = <&vcc5v0_sys>;
-		vcc11-supply = <&vcc_2v0_pldo_s3>;
+		vcc11-supply = <&vcc5v0_sys>;
 		vcc12-supply = <&vcc5v0_sys>;
-		vcc13-supply = <&vdd2_ddr_s3>;
-		vcc14-supply = <&vdd2_ddr_s3>;
+		vcc13-supply = <&vcc5v0_sys>;
+		vcc14-supply = <&vcc5v0_sys>;
 		vcca-supply = <&vcc5v0_sys>;
 
 		gpio-controller;
diff --git a/include/dt-bindings/soc/rockchip,boot-mode.h b/include/dt-bindings/soc/rockchip,boot-mode.h
index 4b0914c09..a0f63a040 100644
--- a/include/dt-bindings/soc/rockchip,boot-mode.h
+++ b/include/dt-bindings/soc/rockchip,boot-mode.h
@@ -10,7 +10,19 @@
 #define BOOT_BL_DOWNLOAD	(REBOOT_FLAG + 1)
 /* enter recovery */
 #define BOOT_RECOVERY		(REBOOT_FLAG + 3)
- /* enter fastboot mode */
+/* reboot by panic */
+#define BOOT_PANIC		(REBOOT_FLAG + 7)
+/* reboot by watchdog */
+#define BOOT_WATCHDOG		(REBOOT_FLAG + 8)
+/* enter fastboot mode */
 #define BOOT_FASTBOOT		(REBOOT_FLAG + 9)
+/* enter charging mode */
+#define BOOT_CHARGING		(REBOOT_FLAG + 11)
+/* enter usb mass storage mode */
+#define BOOT_UMS		(REBOOT_FLAG + 12)
+/* reboot system quiescent */
+#define BOOT_QUIESCENT		(REBOOT_FLAG + 14)
+/* reboot by panic and capture ramdump in uboot through usb */
+#define BOOT_WINUSB		(REBOOT_FLAG + 15)
 
 #endif
